File: src/main/kotlin/ua/com/lavi/usergenerator/factory/MaleUserFactory.kt
package ua.com.lavi.usergenerator.factory

/**
 * Created by Oleksandr Loushkin
 */

class MaleUserFactory : UserFactory() {

    override fun getLastname(): String {
        return randomStringFromList(readResourceLines("malelastnames.txt"))
    }

    override fun getName(): String {
        return randomStringFromList(readResourceLines("malenames.txt"))
    }

    override fun getPatronomyc(): String {
        return randomStringFromList(readResourceLines("malepatronymics.txt"))
    }
}

File: src/main/scala/ua/com/lavi/usergenerator/factory/MaleUserFactory.scala
package ua.com.lavi.usergenerator.factory

/**
 * Created by Oleksandr Loushkin
 */

class MaleUserFactory extends UserFactory {

    override def getLastname: String = {
        randomStringFromList(readResourceLines("malelastnames.txt"))
    }

    override def getName: String = {
        randomStringFromList(readResourceLines("malenames.txt"))
    }

    override def getPatronomyc: String = {
        randomStringFromList(readResourceLines("malepatronymics.txt"))
    }
}
